spyder_unittest/unittestplugin.py

```diff
@@ -24,16 +26,67 @@
 _ = get_translation("unittest", dirname="spyder_unittest")
 
 
-class UnitTestPlugin(UnitTestWidget, SpyderPluginMixin):
-    """Unit testing."""
+class UnitTestWidgetInSpyder(UnitTestWidget):
+    """
+    Unit test widget for use inside Spyder.
+
+    This class overrides some functions in `UnitTestWidget` to provide better
+    integration with Spyder.
+    """
+
+    def __init__(self, main):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        main : MainWindow
+            Spyder main window
+        """
+        UnitTestWidget.__init__(self, parent=main)
```

Comment:
Widgets must not have references to the main window. This is reserved only to plugins.

The idea of our architecture is for widgets to be isolated of the main window and emit signals in case they need to access it. And for plugins to receive those signals and access properties of the main window.